src/apis/impl/DataOwnerApiImpl.ts

```diff
@@ -65,52 +67,56 @@ export class DataOwnerApiImpl implements DataOwnerApi {
     return dataOwnerId
   }
 
-  async initCryptoFor(
-    user: User,
-    overwriteExistingKeys: boolean = false,
-    keyPair?: { publicKey: string; privateKey: string }
-  ): Promise<{ publicKey: string; privateKey: string }> {
+  async initCryptoFor(user: User, overrideExistingKeys: boolean = false, keyPair?: { publicKey: string; privateKey: string }): Promise<void> {
     const dataOwnerId = this.getDataOwnerIdOf(user)
-
-    const { publicKey, privateKey } = keyPair ?? (await this._generateKeyPair())
-    const jwk = {
-      publicKey: spkiToJwk(hex2ua(publicKey)),
-      privateKey: pkcs8ToJwk(hex2ua(privateKey)),
-    }
-    await this.cryptoApi.cacheKeyPair(jwk)
-    await this.keyStorage.storeKeyPair(`${dataOwnerId}.${publicKey.slice(-32)}`, jwk)
-
     const dataOwner = await this.cryptoApi.getDataOwner(dataOwnerId).catch((e) => {
       throw this.errorHandler.createErrorFromAny(e)
     })
+
     if (dataOwner == null) {
       throw this.errorHandler.createErrorWithMessage(
         `The current user is not a data owner. You must be either a patient, a device or a healthcare professional to call this method.`
       )
     }
 
-    if (dataOwner.dataOwner.publicKey == undefined) {
-      await this._updateUserToAddNewlyCreatedPublicKey(user, dataOwner.dataOwner, publicKey)
-    } else if (dataOwner.dataOwner.publicKey != publicKey && overwriteExistingKeys) {
-      console.log(`Adding a new RSA Key Pair to user ${user.id}`)
-      await this.cryptoApi.addRawKeyPairForOwner(this.maintenanceTaskApi, UserMapper.toUserDto(user)!, dataOwner, {
-        publicKey: publicKey,
-        privateKey: privateKey,
-      })
-    }
+    const userKeyPairs = await this._loadCurrentUserKeyPairsOnDevice(dataOwner.dataOwner)
+    const shouldGenerateOrUseProvidedKeyPair = userKeyPairs === undefined || userKeyPairs.length === 0
```

Comment:
I find the name is not quite clear: 
- If it is true, it means we have no key and we should generate one if and only if the argument `overrideExistingKeys` is true right ? Actually, those two names together are not quite clear; 
- If it is false, it means we already have existing keys and we need to use them;
Am I correct ? 
I think you can just rename it `hasExistingKeys` and switch your condition